variables/src/main.rs

```diff
@@ -0,0 +1,70 @@
+// Constants can be placed outside functions in the global scope, available program wide.
+// They're also 'in-lined' at compile time adding to the speed of Rust.
+const WARP_FACTOR: f64 = 9.98;
+fn main() {
+    // Variables in Rust are immutable by default.
+    // You can however make a variable mutable by adding the 'mut' keyword.
+    // You are not required to type let statements when the compiler can figure out the type
+    // similar to typescript .. Below we type it anyway.
+    let missiles: i32 = 8;
+    let ready: i32 = 2;
+    // Another even stricter variable declaration is the constant.
+    // Typing is required on constants. and Screaming SnakeCase used by convention.
+
+    println!(
+        "Firing {} of my {} missiles (which travel at {} warp)!",
+        ready, missiles, WARP_FACTOR,
+    );
+
+    // You can destructure variables
+    let (silly, dumb, dooda) = ("LOL", "snakes", "sherpa");
+
+    // Variable Shadowing
+    // This is likely a Rust concept that looks a lot like what you see in
+    // javascript for instance, but they make a point to call it 'shadowing'.
+    // But it's the way to accomplish reassignment, etc.
+    let d = 10;
+    {
+        let d = 50;
+        // In this block d will be 50
+    }
+    // In the outter scope d will be 10
+
+    let mut doggie = "woof!";
+    let doggie = "bark!";
+    // doggie will now be immutable 'bark!'
+
```

Comment:
You never used the value of doggie from line 33 anywhere!

What I was demonstrating in the lecture was how to use shadowing to "change" the mutability of a variable (really, you're replacing the variable with another one, but there's no runtime cost so it's effectively changing it from our point of view).

```suggestion
    let mut doggie = "woof!";
    let doggie = doggie;
    // doggie will now be immutable 'woof!'

```